hashintel/hash
# Implement read-only mode for resources & entity-editor

## 🌟 What is the purpose of this PR?

This PR improves the readonly logic in app. Now a page or an entity is readonly (not editable), if they are not created by that user, or not owned by one of the organizations which that user is a member of.


## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543021352041/1203780222636483/f) _(internal)_

## 🔍 What does this change?

- Implement readonly mode for entity editor
- Add `useIsReadonlyModeForResource` hook
- Rename `useIsReadonlyMode` -> `useIsReadonlyModeForApp`

## 🐾 Next steps

- Implement readonly mode for entity-type-editor [Asana task](https://app.asana.com/0/1203179076056209/1203623179643234/f) _(internal)_

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Try editing an entity which is not owned by the logged in user (it should be readonly)
2. Try editing a page which is not owned by the logged in user  (it should be readonly)

## 📹 Demo


https://user-images.githubusercontent.com/39553853/213861903-f3c713c7-392b-4982-a9ec-b40693e953cf.mov
